src/modules/_[spaceId]/HomePage/HomePageNavbar.tsx

```diff
@@ -0,0 +1,140 @@
+import Button from '@/components/Button'
+import Input from '@/components/inputs/Input'
+import useWrapInRef from '@/hooks/useWrapInRef'
+import { cx } from '@/utils/class-names'
+import { useEffect, useRef, useState } from 'react'
+import { BsXCircleFill } from 'react-icons/bs'
+import { HiMagnifyingGlass } from 'react-icons/hi2'
+
+export type HomePageNavbarProps = {
+  logo: JSX.Element
+  auth: JSX.Element
+  colorModeToggler: JSX.Element
+  search: string
+  setSearch: (search: string) => void
+  setFocusedElementIndex: React.Dispatch<React.SetStateAction<number>>
+  onUpClick: () => void
+  onDownClick: () => void
+}
+
+export default function HomePageNavbar({
+  auth,
+  colorModeToggler,
+  logo,
+  search,
+  setSearch,
+  onDownClick,
+  onUpClick,
+}: HomePageNavbarProps) {
+  const [openSearch, setOpenSearch] = useState(false)
+  const searchRef = useRef<HTMLInputElement | null>(null)
```

Comment:
A lot of logic related to Search in the common Navbar component. Please extract all Search logic to separate component and just put it here

Reply:
But its not common navbar component, its only for home page navbar